lib/rmt/downloader.rb

```diff
@@ -4,34 +4,47 @@
 require 'fiber'
 require 'rmt'
 require 'rmt/config'
-require 'rmt/http_request'
+require 'rmt/fiber_request'
 require 'rmt/deduplicator'
 
 class RMT::Downloader
 
-  class Exception < RuntimeError
-  end
+  class Exception < RuntimeError; end
+  class NotModifiedException < RuntimeError; end
 
-  attr_accessor :repository_url, :local_path, :concurrency, :logger, :auth_token
+  attr_accessor :repository_url, :local_path, :concurrency, :logger, :auth_token, :cache_path
 
-  def initialize(repository_url:, local_path:, auth_token: nil, logger: nil)
+  def initialize(repository_url:, local_path:, auth_token: nil, logger: nil, cache_path: nil)
     Typhoeus::Config.user_agent = "RMT/#{RMT::VERSION}"
     @repository_url = repository_url
     @local_path = local_path
     @concurrency = 4
     @auth_token = auth_token
     @logger = logger || Logger.new('/dev/null')
+    @cache_path = cache_path
   end
 
-  def download(remote_file, checksum_type = nil, checksum_value = nil)
+  def download(remote_file, checksum_type: nil, checksum_value: nil, use_cache: false)
```

Comment:
Why not make `use_cache: true` the default behavior? That would save you from explicitly setting it with each call to this method.

Reply:
I was thinking that it would make it explicit that cache is being used. I could just enable checking cache whenever `cache_path` is set.

Reply:
Up to you!

Reply:
But yeah, at the moment caching should be used for all `downloader.download` calls, so I've just enabled it by default.